src/cpp/session/modules/SessionDataImportV2.R

```diff
@@ -493,3 +502,18 @@
       return(list(error = e))
    })
 })
+
+.rs.addFunction("preview_data_import_async", function(jsonData)
+{
+   tryCatch({
+      params = jsonlite::fromJSON(jsonData)
```

Comment:
Before you invest any time in the socket based approach we should make sure that you are solving a real performance problem. All of htmlwidgets is based on writing and reading many large temp files and it all ends up happening in "user time" (less than 50ms). To me we'd want the socket if we got into the dozens of megabytes of data but not much before that.